### PR TITLE
numa.numa_stress: remove memoryfile to release memory

### PR DESCRIPTION
**Pull request by huangyum**
Remove the memoryfile generated in last round to release memory.
ID: 1167596
Signed-off-by: Yumei Huang <yuhuang@redhat.com>

### CODE REVIEW

**Comment by weliao** [comment]
LGTM
Acked-by: Wei Liao weliao@redhat.com

File: qemu/tests/numa_stress.py [hunk 1]
```diff
@@ -87,6 +87,8 @@ def run(test, params, env):
     most_used_node, memory_used = max_mem_map_node(host_numa_node, qemu_pid)
 
     for test_round in range(test_count):
+        if os.path.isfile(memory_file):
```
Comment (CongLi) [inline qemu/tests/numa_stress.py]: memory_file has been created here.

Comment (yanan-fu) [inline qemu/tests/numa_stress.py]: Agree with CongLi, in my test: the value of "os.path.isfile(memory_file)" is "False" all the time.
I think this file is created in line 105:
 utils.system(numa_dd_cmd, timeout=dd_timeout)

Comment (huangyum) [inline qemu/tests/numa_stress.py]: The file will be created after first round.  
@yanan-fu how many numa nodes are there in your host ?  I tested with my host which have two nodes and "os.path.isfile(memory_file)"  is true in the second round.

Comment (CongLi) [inline qemu/tests/numa_stress.py]: Hi @huangyum 

This error is copied from your comment in bug, it's only round 0.
ERROR 3-smp_16.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.numa.numa_stress -> TestError: Don't have enough memory to execute this test after 0 round

Comment (huangyum) [inline qemu/tests/numa_stress.py]: @CongLi   I tested in @yanan-fu 's host with smaller memory for guest, and it passed.   It turns out that it is affected by the host free memory.  I will add some code before the test. Thanks for your review.

Comment (yanan-fu) [inline qemu/tests/numa_stress.py]: Hi @huangyum 
    In my test, host has 4 nodes, and case failed in round 0. So os.path.isfile(memory_file) is False.
    Now, i think the comment 4 in the bug is not a bug.
    If the case turn ERROR as " Don't have enough memory to execute this test after 0 round", it should be the test environment problem.
    What do u think?

Comment (CongLi) [inline qemu/tests/numa_stress.py]: Cool~ thanks!

Comment (huangyum) [inline qemu/tests/numa_stress.py]: @yanan-fu  I also hit the issue " Don't have enough memory to execute this test after 1 round" in my host with two numa nodes.  The "os.remove"  fix it  as the memory_file consumed much memory in host.

Comment (huangyum) [inline qemu/tests/numa_stress.py]: @yanan-fu And yes, the comment 4 in the bz is not a bug literally. It's about the host env problem.  I suggest to run the case with suitable memory.  Thanks for your test!

**Comment by huangyum** [comment]
@CongLi  @yanan-fu  could you please help review again? I add some more codes. Thanks :)

File: qemu/tests/numa_stress.py [hunk 2]
```diff
@@ -79,6 +79,9 @@ def run(test, params, env):
     memory_file = utils_misc.get_path(tmpfs_path, "test")
     dd_cmd = "dd if=/dev/urandom of=%s bs=1k count=%s" % (memory_file,
                                                           tmpfs_size)
+    if utils_memory.freememtotal() < tmpfs_size:
+        raise error.TestNAError("Host does not have enough free memory to run the test, "
+                                "skipping test...")
```
Comment (yanan-fu) [inline qemu/tests/numa_stress.py]: guest with large memory:
(1/1) smp_8.8196m.repeat1.Host_RHEL.m6.u8.spice.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.numa.numa_stress: SKIP

Guest with small memory:
(1/1) smp_2.2048m.repeat1.Host_RHEL.m6.u8.spice.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.numa.numa_stress: PASS (2872.02 s)

LGTM
Acked by: Yanan Fu yfu@redhat.com

File: qemu/tests/numa_stress.py [hunk 3]
```diff
@@ -79,6 +79,9 @@ def run(test, params, env):
     memory_file = utils_misc.get_path(tmpfs_path, "test")
     dd_cmd = "dd if=/dev/urandom of=%s bs=1k count=%s" % (memory_file,
                                                           tmpfs_size)
+    if utils_memory.freememtotal() < tmpfs_size:
```
Comment (CongLi) [inline qemu/tests/numa_stress.py]: I think it's better to add step for releasing mem here.
echo 3 > /proc/sys/vm/drop_caches

**Comment by CongLi** [comment]
LGTM,
Acked-by: Cong Li <coli@redaht.com>

**Comment by CongLi** [comment]
@xutian 
Any comments for it?

**Comment by xutian** [comment]
Give me some minis, and I will review it when free. Thanks

File: qemu/tests/numa_stress.py [hunk 4]
```diff
@@ -87,6 +92,9 @@ def run(test, params, env):
     most_used_node, memory_used = max_mem_map_node(host_numa_node, qemu_pid)
 
     for test_round in range(test_count):
+        if os.path.isfile(memory_file):
```
Comment (xutian) [inline qemu/tests/numa_stress.py]: os.path.exists(memory_file) looks more readable

**Comment by huangyum** [comment]
@xutian Updated. Thanks!

**Comment by xutian** [comment]
Merged, thanks @huangyum @yanan-fu @weliao and @CongLi

**Pull request merged**